Added support for "spack load" in spack schema

Mariamajib opened the pull request:
- [x] Update [spack schema](https://github.com/buildtesters/buildtest/blob/devel/buildtest/schemas/spack.schema.json) to add support for load
- [x] Add implementation of emitting spack load command in https://github.com/buildtesters/buildtest/blob/devel/buildtest/builders/spack.py. 
- [x] Add example buildspec to demonstrate this example in https://github.com/buildtesters/buildtest/tree/devel/examples/spack
- [x] Add valid and invalid example for testing with regression test see https://github.com/buildtesters/buildtest/tree/devel/buildtest/schemas/examples/spack.schema.json please put a valid and invalid example so we can test this works during regression test
- [x]  Add documentation in https://buildtest.readthedocs.io/en/devel/buildspecs/spack.html add this at bottom of file

Mariamajib committed:
added spack load to  buildtest/buildtest/schemas/spack.schema.json

Mariamajib committed:
updated buildtest/buildtest/schemas/spack.schema.json

Mariamajib committed:
Added example buildspec "spack_load.yml" in buildtest/examples/spack to demonstrate the example

Mariamajib committed:
Added example buildspec "spack_load.yml" in buildtest/examples/spack to demonstrate the example

Mariamajib committed:
Added invalid spack_load example in buildtest/schemas/examples/spack.schemas.json/invalid

Mariamajib committed:
Added valid spack_load example in buildtest/schemas/examples/spack.schemas.json/valid

Mariamajib committed:
Added implementation of spack load in method "generate_script" in buildtest/builders/spack.py

Mariamajib committed:
updated method "generate_script" in buildtest/builders/spack.py

Mariamajib committed:
updated method "generate_script" in buildtest/builders/spack.py

shahzebsiddiqui commented on buildtest/schemas/examples/spack.schema.json/invalid/spack_load_invalid_example.yml:
```diff
@@ -0,0 +1,16 @@
+        specs: [m4]
+      test:
+        run:
+          specs: ['m4', 1]
```
remove the `1` from here instead you should put the `1` under `specs` section for `load`. You want to make test invalid and it needs to be something specific to the `load` section.

Mariamajib committed:
Added valid spack_load example in buildtest/schemas/examples/spack.schemas.json/valid

Mariamajib committed:
Added support for deactivate in buildtest/schemas/spack.schemas.json

Mariamajib committed:
updated buildtest/schemas/spack.schemas.json

Mariamajib committed:
Added support for deactivate in buildtest/schemas/spack.schemas.json

Mariamajib committed:
updated buildtest/schemas/spack.schemas.json

Mariamajib committed:
Merge branch 'buildtesters:devel' into Mariam_patch

Mariamajib committed:
Added buildtest build and buildtest inspect query in buildtest/tools/docs.py

Mariamajib committed:
Merge branch 'buildtesters:devel' into Mariam_patch

Mariamajib committed:
Added generated files and documented them

Mariamajib committed:
updated style check

Mariamajib committed:
updated style check

Mariamajib committed:
updated spack.rst

Mariamajib committed:
updated spack.rst

shahzebsiddiqui committed:
update documentation for spack load example
remove the output from the `buildtest inspect query` command when showing results for the spack load example
